src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonEncodingStreamWrapper.cs

```diff
@@ -223,6 +223,48 @@ public override int Read(byte[] buffer, int offset, int count)
             }
         }
 
+        public override int Read(Span<byte> destination)
+        {
+            try
+            {
+                if (_byteCount == 0)
+                {
+                    if (_encodingCode == SupportedEncoding.UTF8)
+                    {
+                        return _stream.Read(destination);
+                    }
+
+                    // No more bytes than can be turned into characters
+                    _byteOffset = 0;
+                    _byteCount = _stream.Read(_bytes, _byteCount, (_chars.Length - 1) * 2);
+
+                    // Check for end of stream
+                    if (_byteCount == 0)
+                    {
+                        return 0;
+                    }
+
+                    // Fix up incomplete chars
+                    CleanupCharBreak();
+
+                    // Change encoding
+                    int charCount = _encoding.GetChars(_bytes, 0, _byteCount, _chars, 0);
+                    _byteCount = Encoding.UTF8.GetBytes(_chars, 0, charCount, _bytes, 0);
+                }
+
+                // Give them bytes
+                int count = Math.Min(_byteCount, destination.Length);
+                new ReadOnlySpan<byte>(_bytes, _byteOffset, count).CopyTo(destination);
```

Comment:
Shouldn't there be a static method that allows for copying to a given span, or an instance method on span that allows for copying from an array?

Reply:
cc: @ahsonkhan, @KrzysztofCwalina

Reply:
It is a nice to have but I am not sure if it would be necessary. I would imagine something like:
span.CopyFrom(array);

CopyFrom(T[] sourceArray) however, internally, would just wrap the array within a span and call the existing CopyTo method.